### PR TITLE
CA-155178: Remove squeeze_xen cache if domain disappear

### PR DESCRIPTION
**Pull request by cheng--zhang**
If domains lived no longer than 5 seconds, it will have big opportunity to have cache in squeeze_xen and will not remove by xenstore watcher since this procedure runs every 5 seconds.

This code change is to add a auto clean function in make_host to let the cache cleanup itself.
Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by Zhengchai** [comment]
HI @jonludlam could we look for someone please have a review for this? this is for SCTX-1960. thanks a lot.

**Comment by djs55** [comment]
It looks good to me!

File: ocaml/xenops/squeeze_xen.ml [hunk 1]
```diff
@@ -135,6 +135,23 @@ module Domain = struct
 			| _  -> debug "Ignoring unexpected watch: %s" path
   end
 
+  let remove_gone_domains_cache xc =
+		let open Xenctrl.Domain_info in
+		let current_domians = Xenctrl.domain_getinfolist xc 0 in
```
Comment (johnelse) [inline ocaml/xenops/squeeze_xen.ml]: Small typo here - `current_domians` should be `current_domains`.

**Comment by cheng--zhang** [comment]
Hi @johnelse 
Thank you for your comments, I already fix the typo, please help to check again. Thanks.

**Comment by johnelse** [comment]
@cheng--zhang just realised this is for tampa-lcm...we can't merge anything into tampa-lcm until we're making a tampa hotfix for that fix. Does this fix apply to trunk, and if so is it in trunk-ring3 already?

**Comment by cheng--zhang** [comment]
Hi @johnelse , because of reconstruct the logic of domain arrive and domain leave in Creedence, the new logic will not have such issue.
So we don't need to merge this to Creedence.
Thanks.

**Comment by cheng--zhang** [comment]
@johnelse This fix haven't merged to trunk-ring3. I will create a pull request for this later.
Thanks.

**Pull request closed without merging**